### ISSUE TITLE
Stripe Payment Taken but No OFN Order Created

### ISSUE THREAD

**Issue by lbwright22** [openfoodfoundation/openfoodnetwork#6912]
## Description

Date: 13th December 2020
Hub: Food Lochaber (UK)
Pending order ID R578827212

7 attempts to complete payment of this order were made by the customer between 10th and 11th of December 2020. All failed.

On 13th December two attempts to complete the Stripe Payment were made at 11:13 pm (one for £6.60 and one for £8.34) show as 'processing'.

11:14 pm 13th December a payment of £6.60 was completed. 

![cartorderpayment](https://user-images.githubusercontent.com/45104035/108347463-6e293100-71d8-11eb-8f37-19f1aa9a01ff.jpg)

This is verified by the Stripe record

![cartpaymentstripe](https://user-images.githubusercontent.com/45104035/108347611-9e70cf80-71d8-11eb-90c3-095a85bee53f.jpg)

The order remains in 'cart' and not 'complete'

## Expected Behavior

After payment has been taken for an order the order should move from pending ('cart') to complete and show under the enterprise's record of orders

## Actual Behaviour

Payment was taken but the order remained in 'cart'

## Steps to Reproduce

1.
2.
3.
4.

## Severity

S1

#6903 
## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

**Comment by lbwright22** [comment]
Closing issue as reported in February 2021. 
Between December 2020 and Feb 2021 extensive work carried out on Stripe payment method.  Reopen only if occurs again.

**Comment by filipefurtad0** [comment]
At first, I thought this could be happening due to the balance due of £0.17 in the posted example, but I think this is not the case: If there is a payment for an order, than the order should always be marked as "complete", even if the payment is not yet captured or even if the payment is insufficient. Does this make sense?

I had a quick check in staging-UK, and I'm not sure this is related to SCA. There are orders with Payment Status "complete" but with Order Status "Payment", for full payments made with cash: 

![image](https://user-images.githubusercontent.com/49817236/108354049-d0862f80-71e0-11eb-9f46-c6e4d225ecef.png)

Not sure how this was generated...

**Comment by eugenichafer** [comment]
On 03/20/2021 There was an order (R734283203) from "Territori de Masies" hub charged on Stripe but not completed. 

<img width="1459" alt="Captura de Pantalla 2021-04-07 a les 12 48 14" src="https://user-images.githubusercontent.com/64070973/113857642-0b4d2280-97a3-11eb-9856-7a566adce733.png">

<img width="1091" alt="Captura de Pantalla 2021-04-06 a les 12 23 26" src="https://user-images.githubusercontent.com/64070973/113857853-4fd8be00-97a3-11eb-888f-8a9d1b40b559.png">

**Comment by mkllnk** [comment]
@eugenichafer I redacted the customer's name and email address from your screenshot. We can look that up via the order number.

**Comment by mkllnk** [comment]
Several cases were mentioned here and I will try get a clearer picture on them:

#### Underpaid order doesn't show as completed - UK R578827212

That was the initially reported issue. Payments failed several times and eventually one succeeded, but with a lower amount than the order's total. Since the payment went through checkout, you would expect the payment still to be displayed as complete. Things I need to find out:

- [ ] Did the order have the `completed_at` field set?
- [ ] Was the order's state `complete`?
- [ ] Do production instances have inconsistent order states in their databases?
- [ ] Do we need to adjust the display on the orders screen?

#### Payment was captured but the order remained in payment state - ES R734283203

The order remains in payment state if something goes wrong during the processing of the payment or completion of the order. Tasks:

- [ ] Find the related error in Bugsnag which should point to the root cause of this failure.
- [ ] Propose a resolution strategy for these cases, e.g. retrying to sync the payment with Stripe or manually approving. Special short list on order page?

**Comment by mkllnk** [comment]
@eugenichafer Did you you do anything to the order? I now see it as complete but with payment due:
![Screenshot from 2021-04-08 16-59-42](https://user-images.githubusercontent.com/3524483/113982248-de286f00-988b-11eb-9923-07f83b0372a3.png)

There are two payments associated. The first one was through Stripe, created on Sun, 21 Mar 2021 09:03:18 AEDT +11:00. That could have been the original checkout date. Then a bank transfer payment was added yesterday, Wed, 07 Apr 2021 21:05:44 AEST +10:00. That could have been an attempt to resolve the situation. But it was voided.

I guess that you can resolve the current problem by manually creating a payment and completing it instead of voiding it. Would that work for you?

**Comment by eugenichafer** [comment]
Yes @mkllnk  It was Territori de Masies hub manager who resolve her problem creating a manual payment once she was able to find the order.

**Comment by mkllnk** [comment]
I did some investigation and found that the two cases, UK and Katuma, are very different.

For the UK issue I proposed a little UX change to the product circle to discover these cases earlier. We may then have a better chance of finding the issue. It has been too long now to find any corresponding issue in the logs.

The cause of the Katuma issue is also not known because Katuma's Bugsnag account had only 7 days retention. I opened a PR to switch to our shared Bugsnag account to have 3 months retention. The shop also created a manual payment to resolve the problem. So we have a workaround for that one.

I'm closing this issue because there's no further lead to find the cause. Let's hope that we are in a better position next time.
